sklearn/tests/test_pipeline.py

```diff
@@ -1111,6 +1114,42 @@ def test_input_features_count_vectorizer():
     pipe.fit(JUNK_FOOD_DOCS, y)
     assert_array_equal(pipe.named_steps.clf.input_features_,
                        ['beer', 'burger', 'coke', 'copyright', 'pizza', 'the'])
-    pipe.set_feature_names(["nonsense_is_ignored"])
+    pipe.get_feature_names(["nonsense_is_ignored"])
     assert_array_equal(pipe.named_steps.clf.input_features_,
                        ['beer', 'burger', 'coke', 'copyright', 'pizza', 'the'])
+
+@pytest.mark.filterwarnings('ignore: Default solver will be changed')  # 0.22
+def test_input_features_nested():
+    pipe = Pipeline(steps=[
+        ('inner_pipe', Pipeline(steps=[('select', SelectKBest(k=2)),
+                                       ('clf', LogisticRegression())]))])
+    iris = load_iris()
+    pipe.fit(iris.data, iris.target)
+    xs = np.array(['x0', 'x1', 'x2', 'x3'])
+    assert_array_equal(pipe.input_features_, xs)
+    mask = pipe.named_steps.inner_pipe.named_steps.select.get_support()
+    assert_array_equal(pipe.named_steps.inner_pipe.named_steps.clf.input_features_, xs[mask])
+    pipe.get_feature_names(iris.feature_names)
+    assert_array_equal(pipe.input_features_, iris.feature_names)
+    assert_array_equal(pipe.named_steps.inner_pipe.named_steps.clf.input_features_,
+                       np.array(iris.feature_names)[mask])
+
+                 
+@pytest.mark.filterwarnings('ignore: Default solver will be changed')  # 0.22
+def test_input_features_meta():
+    ovr = OneVsRestClassifier(Pipeline(steps=[('select', SelectKBest(k=2)),
+                                              ('clf', LogisticRegression())]))
```

Comment:
also a test with just a logistic regression in the ovr (so this has no get_feature_names)